### DISCUSSION TITLE
[angular][elsa3] Connect to eventBus

### DISCUSSION THREAD

**Discussion by lemac1** [elsa-workflows/elsa-core#4198] [Q&A]
Hello, I want to create plugin as separate class in my angular project (I assume this is the only way now). In plugin  I need to subscribe to event of elsa studio (for authentication and other purposes). In V2 it was simple as eventBus was exposed in component, how can achieve it in V3?

**Comment by sfmskywalker** [comment]
Hi @lemac1,

In V3, you can still achieve this by leveraging the Container singleton from `typedi`. Here's a quick example to help you get started:

```typescript
import { Container } from "typedi";
import { EventBus } from "@elsa-workflows/elsa-workflows-designer";

class MyPlugin {
    async initialize() {
      // Obtain an instance of the event bus:
      const eventBus = Container.get(EventBus);
    }
}
```

This approach allows you to resolve an instance of the event bus and use it as needed. If you have any further questions or run into any issues, please don't hesitate to ask!
